compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/DesktopPopup.desktop.kt

```diff
@@ -42,45 +51,97 @@ fun rememberCursorPositionProvider(
     alignment: Alignment = Alignment.BottomEnd,
     windowMargin: Dp = 4.dp
 ): PopupPositionProvider = with(LocalDensity.current) {
-    val pointerPositions = LocalPointerPositions.current
-    val cursorPoint = remember { pointerPositions.entries.first().value.round() }
+    val windowMarginPx = windowMargin.roundToPx()
     val offsetPx = IntOffset(offset.x.roundToPx(), offset.y.roundToPx())
+    val cursorPosition = rememberCursorPosition()
+
+    PopupPositionProviderAtOffset(
+        anchorPx = cursorPosition,
+        isAnchorRelative = false,
+        offsetPx = offsetPx,
+        alignment = alignment,
+        windowMarginPx = windowMarginPx
+    )
+}
+
+/**
+ * A [PopupPositionProvider] that positions the popup at the given position relative to the anchor.
+ *
+ * @param positionPx the offset, in pixels, relative to the anchor, to position the popup at.
+ * @param offset [DpOffset] to be added to the position of the popup.
+ * @param alignment The alignment of the popup relative to desired position.
+ * @param windowMargin Defines the area within the window that limits the placement of the popup.
+ */
+@Composable
+fun popupPositionProviderAtPosition(
```

Comment:
Let's make it Experimental or InternalApi

Reply:
It also can't be experimental, because if we use something experimental in the other module (in material from foundation), we can't change it after that because we will break backward compatibility.

As alternative, we can copy-paste it as a temporary measure. Or just stabilize the function right now, which is less good option here.

Regarding the function itself - it looks fine and useful.

Reply:
One note - we should name it as `rememberPopupPositionProviderAtPosition` (see the [guidelines](https://android.googlesource.com/platform/frameworks/support/+/androidx-main/compose/docs/compose-api-guidelines.md#naming-composable-functions-that-remember-the-objects-they-return).

Reply:
> One note - we should name it as `rememberPopupPositionProviderAtPosition`

But this function **doesn't** remember anything.

In fact `rememberCursorPositionProvider` (the version that uses the global pointer) also doesn't remember the value it returns. It does remember the pointer position, but not the popup position provider.

Reply:
Yes, there is another my comment, suggesting to add `remember`. I am not aware of other functions that return a new value every time we recompose.

I don't have a strong opnion, without remember it also works and looks fine. Just probably unusual.

Reply:
By the guide, we can have `defaultStyle`, so I was just wrong :). Let's keep without remember then

Reply:
I made it `remember`, and renamed it accordingly.